csharp/ql/src/semmle/code/csharp/PrintAst.qll

```diff
@@ -504,6 +539,66 @@ final class TypeParametersNode extends PrintAstNode, TTypeParametersNode {
   UnboundGeneric getUnboundGeneric() { result = unboundGeneric }
 }
 
+/**
+ * A node representing the base types of a `ValueOrRefType`.
+ */
+final class BaseTypesNode extends PrintAstNode, TBaseTypesNode {
+  ValueOrRefType valueOrRefType;
+
+  BaseTypesNode() { this = TBaseTypesNode(valueOrRefType) }
+
+  override string toString() { result = "(BaseTypes)" }
+
+  override Location getLocation() { none() }
+
+  override BaseTypeNode getChild(int childIndex) {
+    childIndex = 0 and
+    result.getBaseType() = valueOrRefType.getBaseClass() and
+    isInterestingBaseType(valueOrRefType.getBaseClass()) and
+    result.getDerivedType() = valueOrRefType
```

Comment:
Since the relevant constraints are already applied to `TBaseTypeNode`, you can remove the `isInterestingBaseType(valueOrRefType.getBaseClass())` conjunct.